File: tools/generate_icons.js
// tools/generate_icons.js
// Generates Android launcher icons + Play Store icon from assets/logo/logo.png
// Uses the sharp library (Node.js) — no Python required

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const logoPath = path.resolve('assets/logo/logo.png');
const outputBase = path.resolve('android/app/src/main/res');

// Android required sizes
const sizes = {
  'mipmap-mdpi':    48,
  'mipmap-hdpi':    72,
  'mipmap-xhdpi':   96,
  'mipmap-xxhdpi':  144,
  'mipmap-xxxhdpi': 192,
};

const BLUE = { r: 0, g: 117, b: 201, alpha: 1 };

async function generateIcon(size, outputPath) {
  const logoSize = Math.round(size * 0.8);
  const offset  = Math.floor((size - logoSize) / 2);

  // Resize logo, then composite onto solid blue background
  const resizedLogo = await sharp(logoPath)
    .resize(logoSize, logoSize, { fit: 'contain', background: BLUE })
    .toBuffer();

  await sharp({
    create: {
      width:      size,
      height:     size,
      channels:   4,
      background: BLUE,
    },
  })
    .composite([{ input: resizedLogo, top: offset, left: offset }])
    .png()
    .toFile(outputPath);

  console.log(`Generated: ${outputPath}  (${size}x${size}px)`);
}

async function main() {
  // Android launcher icons
  for (const [folder, size] of Object.entries(sizes)) {
    const dir = path.join(outputBase, folder);
    fs.mkdirSync(dir, { recursive: true });
    await generateIcon(size, path.join(dir, 'ic_launcher.png'));
  }

  // Play Store 512x512
  const playStorePath = path.resolve('assets/logo/play_store_icon.png');
  await generateIcon(512, playStorePath);
  console.log('\nDone! Play Store icon saved to assets/logo/play_store_icon.png');
}

main().catch(err => { console.error(err); process.exit(1); });

File: tools/generate_icons.py
# tools/generate_icons.py
from PIL import Image, ImageDraw
import os

# Icon sizes required for Android
sizes = {
    "mipmap-mdpi": 48,
    "mipmap-hdpi": 72,
    "mipmap-xhdpi": 96,
    "mipmap-xxhdpi": 144,
    "mipmap-xxxhdpi": 192,
}

# Play Store icon
play_store_size = 512

# Source logo already has the correct #0075C9 blue background
logo_path = "assets/logo/logo.png"
output_base = "android/app/src/main/res"

def generate_icon(size, output_path):
    # Load source logo (already has blue background)
    logo = Image.open(logo_path).convert("RGBA")

    # Create a fresh blue background at target size
    background = Image.new("RGBA", (size, size), (0, 117, 201, 255))

    # Resize the logo to 80% of icon size to preserve padding
    logo_size = int(size * 0.8)
    logo_resized = logo.resize((logo_size, logo_size), Image.LANCZOS)

    # Center on background
    offset = (size - logo_size) // 2
    background.paste(logo_resized, (offset, offset), logo_resized)

    # Save as RGB PNG (no alpha needed for launcher icons)
    os.makedirs(os.path.dirname(output_path), exist_ok=True)
    background.convert("RGB").save(output_path, "PNG")
    print(f"Generated: {output_path}  ({size}x{size}px)")

# Generate Android launcher icons
for folder, size in sizes.items():
    output_path = f"{output_base}/{folder}/ic_launcher.png"
    generate_icon(size, output_path)

# Generate Play Store icon
generate_icon(play_store_size, "assets/logo/play_store_icon.png")
print("\nDone! Play Store icon saved to assets/logo/play_store_icon.png")
